presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -213,7 +230,11 @@ public VerificationResult run()
             // Run test queries
             ListenableFuture<Optional<QueryResult<V>>> testQueryFuture = executor.submit(() -> runMainQuery(testQueryBundle.getQuery(), TEST, testQueryContext));
             controlQueryResult = getFutureValue(controlQueryFuture);
-            if (!skipControl) {
+            if (QUERY_BANK_MODE.equals(runningMode)) {
+                controlQueryContext.setState(QueryState.SUCCEEDED);
+                controlQueryContext.setMainQueryStats(EMPTY_STATS);
+            }
+            else if (!skipControl) {
```

Comment:
same comment as before.
This doesn't seem like a good object oriented way of adding Snapshot functionality.
Can we have a Snapshot*Verification class which does this.

Reply:
controlQueryContext is local to AbstractVerification. It can't be accessed in any derived classes.

Reply:
You could work around it by having an abstract method which takes controlQueryContext as an arg such as
`setControlQueryContext(QueryContext controlQueryContext)`  and each flavor of execution chooses how to deal with it.
Having said that, I think is a bigger refactoring and might not be in the scope of this PR. 
We can do it as a later follow up.